modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EChain.java

```diff
@@ -247,74 +247,115 @@ private void analyzeRead() {
         actual = last.after;
     }
 
+    /**
+     * Handles writing byte code for variable/method chains for all given possibilities
+     * including String concatenation, compound assignment, regular assignment, and simple
+     * reads.  Includes proper duplication for chained assignments and assignments that are
+     * also read from.
+     *
+     * Example given 'x[0] += 5;' where x is an array of shorts and x[0] is 1.
+     * Note this example has two links -- x (LVariable) and [0] (LBrace).
+     * The following steps occur:
+     * 1. call link{x}.write(...) -- no op [...]
+     * 2. call link{x}.load(...) -- loads the address of the x array onto the stack [..., address(x)]
+     * 3. call writer.dup(...) -- dup's the address of the x array onto the stack for later use with store [..., address(x), address(x)]
+     * 4. call link{[0]}.write(...) -- load the array index value of the constant int 0 onto the stack [..., address(x), address(x), int(0)]
+     * 5. call link{[0]}.load(...) -- load the short value from x[0] onto the stack [..., address(x), short(1)]
+     * 6. call writer.writeCast(there) -- casts the short on the stack to an int so it can be added with the rhs [..., address(x), int(1)]
+     * 7. call expression.write(...) -- puts the expression's value of the constant int 5 onto the stack [..., address(x), int(1), int(5)]
+     * 8. call writer.writeBinaryInstruction(operation) -- writes the int addition instruction [..., address(x), int(6)]
+     * 9. call writer.writeCast(back) -- convert the value on the stack back into a short [..., address(x), short(6)]
+     * 10. call link{[0]}.store(...) -- store the value on the stack into the 0th index of the array x [...]
+     */
     @Override
     void write(MethodWriter writer) {
-        // can cause class cast exception among other things at runtime
-        writer.writeDebugInfo(location);
-
```

Comment:
I think we want to restore the debug information here? since it can hit classcastexception in other cases, its safe to just do it always?